qemu/tests/win_virtio_driver_install_from_update.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import time
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest.utils_windows import wmic
+
+
+QUERY_TIMEOUT = 360
+INSTALL_TIMEOUT = 360
+OPERATION_TIMEOUT = 120
+
+
+def _pnpdrv_info(session, name_pattern, props=None):
+    cmd = wmic.make_query("path win32_pnpsigneddriver",
+                          "DeviceName like '%s'" % name_pattern,
+                          props=props, get_swch=wmic.FMT_TYPE_LIST)
+    return wmic.parse_list(session.cmd(cmd, timeout=QUERY_TIMEOUT))
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Install drivers from windows update:
+
+    1) Boot guest with the device.
+    2) Check windows update service is running
+       if not, start the service.
+    3) Install drivers from windows update.
+    4) Run driver signature check command in guest.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def uninstall_driver(session, driver_name, device_name, device_hwid):
+        """
+        Uninstall driver.
+
+        :param session: The guest session object.
+        :param driver_name: driver name.
+        :param device_name: device name.
+        :param device_hwid: device hardware id.
+        """
+        logging.info("Uninstalling previous installed driver")
+        for inf_name in _pnpdrv_info(session, device_name, ["InfName"]):
+            uninst_store_cmd = "pnputil /f /d %s" % inf_name
+            status, output = session.cmd_status_output(uninst_store_cmd,
+                                                       inst_timeout)
+            if status:
+                test.error("Failed to uninstall driver '%s' from store, "
+                           "details:\n%s" % (driver_name, output))
+
+        uninst_cmd = "%s remove %s" % (devcon_path, device_hwid)
+        status, output = session.cmd_status_output(uninst_cmd, inst_timeout)
+        # acceptable status: OK(0), REBOOT(1)
+        if status > 1:
+            test.error("Failed to uninstall driver '%s', details:\n"
+                       "%s" % (driver_name, output))
```

Comment:
Highly overlapped with https://github.com/autotest/tp-qemu/pull/2482/files#diff-b6094f287a31dd775ff73923834113dbe30a4b3beb38e93ab3e848035d74fb1bR64, right?

Reply:
yes ,  but could we go ahead with patch first for need to add new steps for patch https://github.com/autotest/tp-qemu/pull/2482  which may need longer time.

Reply:
Sorry @menli820, that is not really accepted for me.

Reply:
Okay, got it, thanks @vivianQizhu